catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -61,11 +68,40 @@
      * @param queryRequest     request used to retrieve the results.
      */
     public ResultIterable(CatalogFramework catalogFramework, QueryRequest queryRequest) {
-        notNull(catalogFramework, "Catalog framework reference cannot be null");
+        this(catalogFramework, queryRequest, 0);
+    }
+
+    /**
+     * Creates an iterable that will call a {@link QueryFunction} to retrieve the results
+     * that match the {@link QueryRequest} provided.
+     *
+     * @param queryFunction reference to the {@link QueryFunction} to call to retrieve the
+     *                      results.
+     * @param queryRequest  request used to retrieve the results.
+     */
+    public ResultIterable(QueryFunction queryFunction, QueryRequest queryRequest) {
+        this(queryFunction, queryRequest, 0);
+    }
+
+    /**
+     * Creates an iterable that will call the {@link CatalogFramework} to retrieve the results
+     * that match the {@link QueryRequest} provided.
+     *
+     * @param catalogFramework reference to the {@link CatalogFramework} to call to retrieve the
```

Comment:
Also 0 is no longer an allowed value (externally). All callers must now pass a positive integer OR use the one that does not take a cap. This way the 0 remains an internal detail